app/src/components/security/RouteGuards.tsx

```diff
@@ -50,240 +52,145 @@ export interface IProjectRoleRouteGuardProps extends RouteProps {
-    return <CircularProgress className="pageProgress" />;
-  }
+  const authStateContext = useAuthStateContext();
 
-  return <>{children}</>;
-};
+  const location = useLocation();
 
-/**
- * Waits for the projectAuthStateContext to finish loading project participant info.
- *
- * Renders a spinner or the `children`.
- *
- * @param {*} { children }
- * @return {*}
- */
-const WaitForProjectParticipantInfo = (props: PropsWithChildren<Record<never, unknown>>) => {
-  const projectAuthStateContext = useContext(ProjectAuthStateContext);
+  useEffect(() => {
+    if (
+      authStateContext.isReady &&
+      !hasAuthParams() &&
+      !authStateContext.auth.isAuthenticated &&
+      !authStateContext.auth.activeNavigator
+    ) {
+      // User is not authenticated and has no active authentication navigator
+      authStateContext.auth.signinRedirect({ redirect_uri: buildUrl(window.location.origin, location.pathname) });
+    }
+  }, [authStateContext.auth, authStateContext.isReady, location.pathname]);
 
-  if (!projectAuthStateContext.hasLoadedParticipantInfo) {
-    // Participant data has not been loaded, can not yet determine if user has sufficient roles
+  if (
+    !authStateContext.isReady ||
+    !authStateContext.simsUserWrapper.isReady ||
+    !authStateContext.auth.isAuthenticated
+  ) {
     return <CircularProgress className="pageProgress" />;
   }
 
-  return <>{props.children}</>;
-};
-
-/**
- * Checks if the user is a registered user or has a pending access request.
- *
- * Redirects the user as appropriate, or renders the `children`.
- *
- * @param {*} { children }
- * @return {*}
- */
-const CheckIfAuthenticatedUser = (props: PropsWithChildren<Record<never, unknown>>) => {
-  const { keycloakWrapper } = useContext(AuthStateContext);
-
-  const location = useLocation();
+  if (!authStateContext.auth.isAuthenticated) {
+    // Redirect to forbidden page
+    return <Redirect to="/forbidden" />;
+  }
 
-  if (!keycloakWrapper?.isSystemUser()) {
+  if (!authStateContext.simsUserWrapper.systemUserId) {
     // User is not a registered system user
-    if (keycloakWrapper?.hasAccessRequest) {
+    if (authStateContext.simsUserWrapper.hasAccessRequest) {
       // The user has a pending access request, restrict them to the request-submitted or logout pages
-      if (location.pathname !== '/request-submitted' && location.pathname !== '/logout') {
+      if (location.pathname !== '/request-submitted') {
         return <Redirect to="/request-submitted" />;
       }
     } else {
-      // The user does not have a pending access request, restrict them to the access-request, request-submitted or logout pages
-      if (!['/access-request', '/request-submitted', '/logout'].includes(location.pathname)) {
+      // The user does not have a pending access request, restrict them to the access-request or request-submitted pages
+      if (!['/access-request', '/request-submitted'].includes(location.pathname)) {
```

Comment:
My only thought with this is that there are other components where the unauth routes are mentioned, so we'd have to come up with a more generalized solution. But then it would feel weird, I think, to have some routes in constants, and others not. But then that gets messy as some of the routes have path variables. So I'm not sure what the best course of action is :/

I guess I'd be inclined to leave it for now, pending some general overall routing enhancement.